configure.ac

```diff
@@ -1519,6 +1519,10 @@ case "$host_cpu-$host_vendor-$host_os" in
     AC_DEFINE([HAVE_MINGW], [1], [Define if using MinGW])
     H5_CPPFLAGS="-D_GNU_SOURCE -D__USE_MINGW_ANSI_STDIO $H5_CPPFLAGS"
     ;;
+  *cygwin*)
+    AC_DEFINE([HAVE_CYGWIN], [1], [Define if using MinGW])
+    H5_CPPFLAGS="-D_GNU_SOURCE"
```

Comment:
Might need  $H5_CPPFLAGS so that it is appeneded (see the linux section above - line 1514)